neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -181,11 +191,57 @@ async def streaming_chat(self, user_input: str,
         :return: Nothing.  Response values are put on a queue whose consumtion is
                 managed by the Iterator aspect of AsyncCollatingQueue on the InvocationContext.
         """
+        # Set up some member variable state so that run_it() can use it
+        self.invocation_context = invocation_context
+        journal: Journal = self.invocation_context.get_journal()
+        self.interceptor = InterceptingJournal(journal, origin=None)
+
+        # Set up the input dictionary that will show up in an Observability/tracing app
+        inputs: Dict[str, Any] = {
+            "user_input": user_input,
+            "sly_data": sly_data,
+            "chat_context": chat_context
+        }
+
+        # Set up configuration for creating the tracing context.
+        # These are the bare minimum required to get output and metadata correct
+        # in the tracing reporting.
+        config: Dict[str, Any] = {
+            "interceptor": self.interceptor,
+            "invocation_context": self.invocation_context
+        }
+
+        # Get a toolkit-specific implementation for creating the tracing context
+        tracing_factory: ContextTypeTracingContextFactory = \
+            MasterTracingContextFactory.create_tracing_context_factory()
+        # For the factory args, we are our own run_target.
+        tracing_context: RunTarget = tracing_factory.create_tracing_context(config, run_target=self)
+
+        # Run the run_target that was given back by the factory.
+        await tracing_context.run_it(inputs)
+
+    async def run_it(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        This method is effectively a callback which is invoked within
+        the tracing context infrastructure for the toolkit/context_type.
+
+        :param inputs: A dictionary with the following keys:
+                "user_input": A string with the user's input
+                "sly_data": A mapping whose keys might be referenceable by agents, but whose
+                             values should not appear in agent chat text. Can be None.
+                "chat_context": A ChatContext dictionary that contains all the state necessary
+                                to carry on a previous conversation, possibly from a different server.
+        :return: A dictionary with the user's input
+        """
+        user_input: str = inputs.get("user_input")
+        sly_data: Dict[str, Any] = inputs.get("sly_data")
+        chat_context: Dict[str, Any] = inputs.get("chat_context")
+
         if self.front_man is None:
-            await self.set_up(invocation_context, chat_context)
+            await self.set_up(self.invocation_context, chat_context)
```

Comment:
Now that we are within the run_it() method, we need to use the invocation_context member we set up earlier.